src/web-components/vf-score.js

```diff
@@ -0,0 +1,137 @@
+// ## Description
+// 
+// This file implements `vf-score`, the web component that acts as the 
+// container for the entire component. `vf-score`'s shadow root holds
+// the HTML element that the renderer renders too.
+// 
+// All actual drawing is called from `vf-score`. 
+
+import Vex from '../index';
+
+export class VFScore extends HTMLElement {
+  constructor() {
+    super();
+
+    // Defaults
+    this.x = 10; // The starting x position of a system within the score.
+    this.y = 0; // The starting y position of system within the score. 
+    this.width = 500; // The entire width of the element holding the music notation. 
+    this.height = 150; // The entire height of the element holding the music notation.
+    this.stavesPerLine = 1; 
+
+    this.systemsAdded = 0; // Counter that keeps track of how many systems have dispatched 
+                           // events signalling that they are ready to be drawn. When the
+                           // number of children matches this counter, the entire score
+                           // is ready to be drawn.
+                           // FOR THIS PR: a vf-score can only have one vf-system. 
+
+    this.attachShadow({ mode:'open' });
+
+    // The 'systemCreated' event is dispatched by a vf-systen when it has finished creating 
+    // and adding its staves. vf-score listens to this event so that it can add that it can 
+    // detect when the vf-system is ready to be drawn.
+    this.addEventListener('systemCreated', this.systemCreated);
+
+    // These events are dispatched by the corresponding elements (ex: vfVoiceReady is dispatched
+    // by a vf-voice) when they are added to the DOM. vf-score listens to these events so that it 
+    // can set the elements' Factory and/or Registry instances, since these are shared by a 
+    // vf-score and all its children to maintain the same render queue. 
+    this.addEventListener('vfVoiceReady', this.setFactory);
+    this.addEventListener('vfStaveReady', this.setFactory);
+    this.addEventListener('vfStaveReady', this.setRegistry);
+    this.addEventListener('vfSystemReady', this.setFactory);
+  }
+
+  connectedCallback() {
+    this.x = parseInt(this.getAttribute('x')) || this.x;
```

Comment:
I would try to read attributes in `attributeChangedCallback`. It can make some initialization trickier since they come in one at a time, but it allows attributes to change, and some ways of constructing elements (maybe from frameworks like React) might trigger connected callback before setting attributes, so it's more robust.